.github/workflows/test-e2e.yml

```
@@ -0,0 +1,104 @@
+name: E2E Tests
+
+on:
+  pull_request:
+
+permissions:
+  id-token: write
+
+jobs:
+  test-e2e:
+    name: Run on Ubuntu
+    runs-on: ubuntu-latest
+    env:
+      AWS_ROLE_ARN: arn:aws:iam::489668258961:role/multi-platform-github-action
+      AWS_REGION: us-east-1
+      AWS_ROLE_DURATION: 900 # 15 minutes
+      INSTANCE_TAG: ${{ github.run_id }}-development
+    steps:
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
+        with:
+          # this might remove tools that are actually needed,
+          # if set to "true" but frees about 6 GB
+          tool-cache: false
+
+          docker-images: false
+
+      - name: Clone the code
+        uses: actions/checkout@v4
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v5.1.1
+        with:
+          role-to-assume: ${{ env.AWS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+          role-duration-seconds: ${{ env.AWS_ROLE_DURATION }}
+
+      - name: Validate AWS Credentials
+        run: |
+          aws sts get-caller-identity
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Install the latest version of kind
+        run: |
+          curl -Lo ./kind https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64
+          chmod +x ./kind
+          sudo mv ./kind /usr/local/bin/kind
+
+      - name: Verify kind installation
+        run: kind version
+
+      - name: Create kind cluster
+        run: kind create cluster
+        env:
+          KIND_EXPERIMENTAL_PROVIDER: podman
+
+      - name: Build and load images
+        env:
+          KIND_EXPERIMENTAL_PROVIDER: podman
+        run: |
+          go mod tidy
+          make load-image
+          podman exec kind-control-plane crictl images
+
+      - name: Install dependencies
+        env:
+          KIND_EXPERIMENTAL_PROVIDER: podman
+        run: |
+          make tekton
+
+      - name: Deploy MPC
+        env:
+          KIND_EXPERIMENTAL_PROVIDER: podman
+        run: |
+          make deploy
+
+      - name: Run Test e2e
+        env:
+          KIND_EXPERIMENTAL_PROVIDER: podman
+        run: |
+          make test-e2e
+
+      # Reconfigure AWS credentials for cleanup to ensure they're available even if
+      # The credentials expired (15-minute duration) during long-running tests
+
+      - name: Configure AWS Credentials for Cleanup
+        if: always()
+        uses: aws-actions/configure-aws-credentials@v5.1.1
+        with:
+          role-to-assume: ${{ env.AWS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+          role-duration-seconds: ${{ env.AWS_ROLE_DURATION }}
+
+      - name: Delete AWS EC2 Instances
+        if: always()
+        run: ./out/devsetup cleanup-instances ${{ env.INSTANCE_TAG }}
+
+      - name: Delete AWS EC2 Key Pair
+        if: always()
+        run: ./out/devsetup cleanup-keypair ${{ github.run_id }}
```

MPC is now connected new coverage collection method in Codecov, to be later also channeled via the [CoverPort project.](https://redhat.enterprise.slack.com/archives/C09MYT9LQCB) . All testing workflows should have this additional step to send their overage data to Codecov. See the [PR](https://github.com/konflux-ci/multi-platform-controller/pull/668) that brought this change to MPC.

```suggestion
        run: ./out/devsetup cleanup-keypair ${{ github.run_id }}
        
      - name: Upload to Codecov
        uses: codecov/codecov-action@v5
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          flags: e2e-tests
```

---

@psturc konflux-ui's [e2e coverage report](https://app.codecov.io/gh/konflux-ci/konflux-ui?flags%5B0%5D=e2e) is looking a bit empty, though the code you pointed to has been pushed 2 weeks ago.

Are you sure this coverage collection isn't more suited to tests from e2e-tests _repository_? 

Also, could your team please provide more documentation about which way to add test coverage tracking with coverport according to the test level added? As you can see in the PR, an e2e test is being introduced to MPC for the first time, the coverage-collecting tool to track its work is part of the PR - we just don't have a lot of data how to actually do it apparently.

---

You know what? I'm not going to put in any more effort than the people responsible for this tool. It's only going to provide the coverage data that they care enough to allow their users to collect.

---

> @psturc konflux-ui's [e2e coverage report](https://app.codecov.io/gh/konflux-ci/konflux-ui?flags%5B0%5D=e2e) is looking a bit empty, though the code you pointed to has been pushed 2 weeks ago.

They don't run e2e tests against **every commit** in the `main` branch (yet). They have a nightly job that runs against `main` branch which means that **not every commit in the main branch** will have e2e-tests coverage data (which is what you currently see).

They do run e2e tests against [every PR though](https://app.codecov.io/gh/konflux-ci/konflux-ui/pulls). So every PR which had run the testing pipeline/script will contain the data about e2e tests coverage.

> Are you sure this coverage collection isn't more suited to tests from e2e-tests repository?

Even running a container without any test running against it will produce coverage data that you can collect. It's test suite independent tool.

> Also, could your team please provide more documentation about which way to add test coverage tracking with coverport according to the test level added?

The first phase of the project was focused on integrating components that are utilizing konflux integration pipeline for deploying the component, running e2e tests and collecting the e2e tests coverage.
We've put together [this AI agent skill](https://github.com/konflux-ci/coverport/blob/main/.claude/skills/coverport-integration/SKILL.md) that does the integration for you.

However we haven't focused on providing a docs/tools to help with integrating coverport with github actions CI. We've got [this example of integration for konflux-ui](https://github.com/konflux-ci/konflux-ui/commit/2d4d40dd0ec5ff910b4b868055998a56486b68bf), but it's not documented yet. I will talk to my team if we can squeeze some work regarding the documentation for these cases for the next sprint.